lib/s3.js

```diff
@@ -0,0 +1,55 @@
+import AWS from 'aws-sdk';
+import B from 'bluebird';
+import { zip } from 'appium-support';
+import { v4 as uuid } from 'uuid';
+
+const Bucket = process.env.AWS_S3_BUCKET;
+
+let s3 = new AWS.S3({
+  params: {
+    Bucket,
+  },
+});
+
+
+
+/**
+ * Zips a local directory and uploads it to S3
+ * @param {*} localDir 
+ */
+async function uploadZip (localDir) {
+  if (!process.env.AWS_S3_BUCKET) {
+    throw new Error('To use S3, you must specify AWS_S3_BUCKET in environment variables');
+  }
+  const base64Zip = await zip.toInMemoryZip(localDir);
+
+  // Return the call to s3.upload as a promise
+  // (NOTE: B.promisify doesn't work for s3.upload, it causes s3/managed_upload.js to throw an error `self.service.constructor.__super__ is not a constructor`)
+  return await new B((resolve, reject) => {
```

Comment:
ok looks good, however i think a comment got lost about promisifying. i said:

2 things:

1. you want `return await` probably
2. you probably want to pass the `context` object for proper binding:

```js
return await B.promisify(s3.upload, {context: s3})({...});
```